modules/apps/dataset/dataset-custom-view-impl/src/main/java/com/liferay/dataset/custom/view/internal/settings/DatasetCustomViewSettingsFactoryImpl.java

```diff
@@ -121,8 +121,20 @@ private String _getLegacyJSONString(
 			_getClayDataSetDisplaySettingsNamespace(
 				httpServletRequest, datasetDisplayId);
 
-		return portalPreferences.getValue(
+		String jsonString = portalPreferences.getValue(
 			clayDataSetDisplaySettingsNamespace, "activeViewSettingsJSON");
+
+		if (jsonString != null) {
+			_datasetCustomViewEntryLocalService.updateDatasetCustomViewEntry(
+				_datasetCustomViewEntryLocalService.
+					createDatasetCustomViewEntry(jsonString));
+
+			portalPreferences.setValue(
```

Comment:
Shouldn't we delete the portalPreferences entry from the DB rather than setting a null value?

Reply:
That's the only formal way I found to delete portlet prefs. However I'll check again...

Reply:
Maybe it's an idiom I'm not aware of, I should have checked it first